homeassistant/components/fmi/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for FMI (Finnish Meteorological Institute) integration."""
+
+import fmi_weather_client as fmi_client
+from fmi_weather_client.errors import ClientError, ServerError
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME, CONF_OFFSET
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from . import base_unique_id
+from .const import _LOGGER, FORECAST_OFFSET
+
+
+async def validate_user_config(hass: core.HomeAssistant, data):
+    """Validate input configuration for FMI.
+
+    Data contains Latitude / Longitude provided by user or from
+    HASS default configuration.
+    """
+    latitude = data[CONF_LATITUDE]
+    longitude = data[CONF_LONGITUDE]
+
+    errors = ""
+
+    # Current Weather
+    try:
+        weather_data = await hass.async_add_executor_job(
+            fmi_client.weather_by_coordinates, latitude, longitude
+        )
+
+        return {"place": weather_data.place, "err": ""}
+    except ClientError as err:
+        err_string = (
+            "Client error with status "
+            + str(err.status_code)
+            + " and message "
+            + err.message
+        )
+        errors = "client_connect_error"
+        _LOGGER.error(err_string)
+    except ServerError as err:
+        err_string = (
+            "Server error with status "
+            + str(err.status_code)
+            + " and message "
+            + err.body
+        )
+        errors = "server_connect_error"
+        _LOGGER.error(err_string)
+
+    return {"place": "None", "err": errors}
+
+
+class FMIConfigFlowHandler(config_entries.ConfigFlow, domain="fmi"):
+    """Config flow handler for FMI."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle user step."""
+        # Display an option for the user to provide Lat/Long for the integration
+        errors = {}
+        if user_input is not None:
+
+            await self.async_set_unique_id(
+                base_unique_id(user_input[CONF_LATITUDE], user_input[CONF_LONGITUDE])
+            )
+            self._abort_if_unique_id_configured()
+
+            valid = await validate_user_config(self.hass, user_input)
+
+            if valid.get("err", "") == "":
```

Comment:
This is a weird way of doing error handling. Please raise on the validation and wrap it with a try.